### PR TITLE
add parser support for udf

### PR DESCRIPTION
**Pull request by chrisxu333**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # 6382

## What this PR does / why we need it: 
add parser support for user-defined function grammar /  query logic handling

### CODE REVIEW

File: test/distributed/cases/function/udf.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+-- @case
+-- @desc:test for drop function (no arg)
+-- @label:bvt
+drop function helloWorld ();
+
+-- @case
+-- @desc:test for drop function (multiple args)
+-- @label:bvt
+drop function charAt (varchar, int);
```
Comment (fengttt) [inline test/distributed/cases/function/udf.sql:22]: can you drop twosum with different param types?   what do we expect, fail or succeed?

File: test/distributed/cases/function/udf.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+-- @case
+-- @desc:test for create function (no arg)
+-- @label:bvt
+create function helloworld () returns int language sql as 'select id from test_table limit 1';
+
+-- @case
+-- @desc:test for create function (multiple args)
+-- @label:bvt
+create function twosum (x int, y int) returns int language sql as 'select $1 + $2' ;
```
Comment (fengttt) [inline test/distributed/cases/function/udf.sql:12]: can we create function with same name but different number/type of params?   

test function name with mixed case, punctuation characters etc.

File: test/distributed/cases/function/udf.sql (hunk 1 above)
Comment (chrisxu333) [reply test/distributed/cases/function/udf.sql:22]: We expect this to be failed, as function is specified by both its name and parameters.

File: test/distributed/cases/function/udf.sql (hunk 2 above)
Comment (chrisxu333) [reply test/distributed/cases/function/udf.sql:12]: Yes we can, sure I'll add this to test.

**Review by daviszhen — APPROVED** [review]

**Review by aressu1985 — COMMENTED** [review]
please move udf.result file to the dir that udf.sql is in

**Comment by chrisxu333** [comment]
> please move udf.result file to the dir that udf.sql is in

Okay

**Review by fengttt — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by iamlinjunhong — APPROVED** [review]

**Pull request merged**